_posts/2024-02-05-cde-customization.adoc

```diff
@@ -0,0 +1,434 @@
+
+NOTE: It's not necessary to use Eclipe Che default image as the base image. I am maintaining https://github.com/devfile/developer-images/[a repository with a list of Dockerfiles for Che CDEs] using popular base images.
+
+### Build and publish the custom image
+
+The Dockerfile can be built locally (on your laptop) and pushed to a remote registry such as Docker Hub or Quay.io. That requires `docker` or `podman` and a good Internet connection (the base image, UDI, has a size of a few Gigabytes).
+
+Another option is to build and push the `Dockerfile` without leaving the `ruby-sample` CDE started in STEP1. `podman` is included in the universal developer image and can be used to build and push the Dockerfile above.
+
+My preferred alternative though, and the one used https://github.com/l0rd/rails-sample/blob/dockerfile/.devfile.Dockerfile[branch `dockerfile` of the ruby-sample repository], is to leverage GitHub actions to automatically build and push the image:
+
+1. Add the Dockerfile to the git repository as `.devfile.Dockerfile`
+2. Add a GitHub workflow to build and push the image automatically (https://github.com/l0rd/rails-sample/blob/dockerfile/.github/workflows/image-build.yaml[see an example here]) 
+3. Push both files to a remote branch of the git repository (that will trigger the GitHub action)
+
+image::/assets/img/cde-customization/github-action.png[]
+{nbsp} +
+As a result the image will be built and pushed automatically at every commit. The https://github.com/l0rd/rails-sample/blob/dockerfile/.github/workflows/image-build.yaml#L29[branch `dockerfile` of the ruby-sample repository] pushes the image to `quay.io/mloriedo/rails-blog-cde:latest`.
+
+NOTE: I have named the file as `.devfile.Dockerfile`: even if any name is ok I like to use the `.devfile` prefix as these images are usually used in devfiles (as we will do STEP 3).
+
+### Start a new CDE using the custom image
+ 
+The simplest way to start a workspace using a new custom image is to use https://eclipse.dev/che/docs/stable/end-user-guide/url-parameter-for-container-image/[`?image=<image-name>` Eclipse Che URL parameter]. 
+
+For example, to start a CDE using the custom image `quay.io/mloriedo/rails-blog-cde` and cloning rails-sample, the following string can be used in the "Create Workspace" page:
+
+[attributes]
+....
+https://github.com/l0rd/rails-sample&image=quay.io/mloriedo/rails-blog-cde
+....
+
+image::/assets/img/cde-customization/image-url-parameter.png[]
+{nbsp} +
+
+Or this URL can be used to start it on Red Hat Developer Sandbox: 
+
+[attributes]
+....
+https://workspaces.openshift.com/#https://github.com/l0rd/rails-sample&image=quay.io/mloriedo/rails-blog-cde
+....
+
+### Verify the new CDE Pod image
+
+The CDE Pod uses a `rails-blog-cde` container rather than Eclipse Che default universal developer image. To verify it open a Terminal from VisualStudio Code and run kubectl:
```

Comment:
```suggestion
The CDE Pod now uses a `rails-blog-cde` container rather than the Eclipse Che default universal developer image. To verify it open a Terminal from Visual Studio Code and run `kubectl`:
```